app/controllers/vendor_api/decisions_controller.rb

```diff
@@ -59,5 +55,20 @@ def invalid_metadata(_e)
                  }],
       }, status: 422
     end
+
+    def repond_to_decision(decision)
+      if decision.save
+        render json: { data: SingleApplicationPresenter.new(application_choice).as_json }
+      else
+        render_error_response(decision.errors)
+      end
+    end
+
+    # Takes a object with ActiveModel::Validations and render the `errors`
+    # as API response.
+    def render_error_response(errors)
+      errors = errors.map { |_key, message| { error: 'BadRequest', message: message } }
```

Comment:
not sure we should reassign the input param`errors`,
should we rename it to `error_messages` (or something)?